action_tutorials/action_tutorials_py/action_tutorials_py/fibonacci_action_server.py

```diff
@@ -100,12 +106,17 @@ def execute_callback(self, goal_handle):
         result.sequence = feedback_msg.sequence
         return result
 
+    def cancel_callback(self, goal_handle):
+        self.get_logger().info('Canceling goal...')
+        return CancelResponse.ACCEPT
+
 
 def main(args=None):
     try:
         with rclpy.init(args=args):
             fibonacci_action_server = FibonacciActionServer()
-            rclpy.spin(fibonacci_action_server)
+            executor = MultiThreadedExecutor()
```

Comment:
use multi-threaded executor here, so that it can take the cancel request concurrently processing the goal request.